### PR TITLE
User can login

### PR DESCRIPTION
**Pull request by raomanasa**
[PT-LINK](https://www.pivotaltracker.com/story/show/171048160)
User story-
As a User
In order to get access my trip package
I would like to log in
Tasks-
- Add j-tockauth.
- Add login button.
- Adds conditional logout button
Screenshot 2020-02-11 at 11 22 55 AM

Screenshot 2020-02-11 at 11 24 27 AM

### CODE REVIEW

File: src/state/store/initialState.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
 const initialState = {
   lat: 0,
   lng: 0,
+  name: null,
+  authenticated: false,
+  displayLoginButton: true,
+  displaySignupButton: true,
+  authMessage: null,
+  userAttrs: null,
+  message: null,
```
Comment (sundqvistjohan) [inline src/state/store/initialState.js]: I don't think we need to store the user attributes and the state variable message already exists and you're not using it so remove this one

File: src/state/store/initialState.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
 const initialState = {
   lat: 0,
   lng: 0,
+  name: null,
```
Comment (sundqvistjohan) [inline src/state/store/initialState.js]: What's name?

File: src/components/Destination.jsx [hunk 3]
```diff
@@ -28,7 +29,7 @@ const Destination = props => {
     e.preventDefault();
     let response = await getCoords(e.target.place.value);
     if (!response.error) {
-      if (response.data.status !== "ZERO_RESULTS") {
+    if (response.data.status !== "ZERO_RESULTS") {
```
Comment (sundqvistjohan) [inline src/components/Destination.jsx]: Undo this formatting change

File: src/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  return (
+    <div id="login">{loginFunction}</div>
+  );
+};
+  const mapStateToProps = state => ({
+    authenticated: state.authenticated,
+    userAttrs: state.userAttrs,
+    authMessage: state.authMessage,
+    displaySignupButton: state.displaySignupButton,
```
Comment (sundqvistjohan) [inline src/components/Login.jsx]: There is no signup functionality

File: src/modules/auth.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import JtockAuth from "j-tockauth";
+
+const auth = new JtockAuth({
+  host: "http://localhost:3000",
+  //host: process.env.REACT_APP_BASEURL,
```
Comment (sundqvistjohan) [inline src/modules/auth.js]: Remove, unless we want to take this opportunity to introduce those environment variables

**Review by sundqvistjohan — CHANGES_REQUESTED** [review]
Remove references to signup button since that functionality doesn't exist. Unless we want to add it now.

File: src/state/store/initialState.js (hunk 1 above)
Comment (raomanasa) [reply src/state/store/initialState.js]: done

File: src/components/Destination.jsx (hunk 3 above)
Comment (raomanasa) [reply src/components/Destination.jsx]: done

File: src/components/Login.jsx (hunk 4 above)
Comment (raomanasa) [reply src/components/Login.jsx]: done

File: src/modules/auth.js (hunk 5 above)
Comment (raomanasa) [reply src/modules/auth.js]: done

File: src/state/store/initialState.js (hunk 2 above)
Comment (raomanasa) [reply src/state/store/initialState.js]: done

File: src/components/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      .catch(error => {
+        props.changeAuthMessage(error);
+      });
+  };
+  
+  let loginFunction;
+  switch (true) {
+    case props.displayLoginButton &&
+      !props.authenticated:
+      console.log("entered 0")
```
Comment (sundqvistjohan) [inline src/components/Login.jsx]: You can remove these console.log statements

File: src/components/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        <Button
+          id="login-button"
+          onClick={() => props.changeLoginButton(false)}
+        >
+          Login
+        </Button>
+      );
+      break;
+    case !props.displayLoginButton && !props.authenticated:
+      console.log("entered 1")
```
Comment (sundqvistjohan) [inline src/components/Login.jsx]: Here as well

File: src/components/Login.jsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            <label>password</label>
+            <input name="password" type="password" id="password"></input>&nbsp;
+            <button id="submit">Submit</button>
+          </form>
+          {props.authMessage}
+        </>
+      );
+      break;
+    case props.authenticated:
+      console.log("entered 2")
```
Comment (sundqvistjohan) [inline src/components/Login.jsx]: And here

Comment (raomanasa) [reply src/components/Login.jsx]: done

File: src/components/Login.jsx (hunk 7 above)
Comment (raomanasa) [reply src/components/Login.jsx]: done

File: src/components/Login.jsx (hunk 6 above)
Comment (raomanasa) [reply src/components/Login.jsx]: done

**Review by sundqvistjohan — APPROVED** [review]

**Review by mantanquzzmo — APPROVED** [review]

**Pull request merged**